### ISSUE TITLE
[FEAT] Enable vector properties

### ISSUE THREAD

**Issue by noamzbr** [deepchecks/deepchecks#1912]
**Is your feature request related to a problem? Please describe.**
Enable user properties that return a vector of numbers, rather than a single number per sample.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by nirhutnik** [comment]
@noamzbr Can you elaborate?
1. Why is this needed? What kind of properties come in a vector?
2. What did you mean regarding distance?
3. In general, task is unclear

**Comment by noamzbr** [comment]
I agree this issue should wait for a proper solution to using embedding
